LibrePhotos/librephotos
# #310 Read/write metadata from/to disk

1. Extract rating when scanning media.
2. Read metadata tags from XMP sidecar files if present (priority order: <basename>.xmp, <basename>.<ext>.xmp, <basename>.ext).
3. Write metadata to XMP sidecar whenever it is changed on a media file (currently only Rating, when the user marks / unmarks as favorite).
4. New CLI command "save_metadata" that saves the metadata for all media files (currently only Rating).
5. Add user setting in db for configuring if / how to save metadata to disk.

How to test:

1. Choose a photo in your collection that is not favorited. Copy the path on the file system.
2. Create an XMP sidecar file with a Rating=5:
    `exiftool path/to/the-chosen-photo.xmp -Rating=5`
3. Scan your photos (filesystem)
4. Check that the chosen photo now shows as favorited (refresh page if needed)
5. Edit "save_metadata_to_disk" in the database for your test user to "SIDECAR_FILE"
6. Mark a photo as favorite
7. Check that a <basename>.xmp file has been added with Rating=4 (or whatever your favorite_min_rating is set to)
8. Edit "save_metadata_to_disk" in the database for your test user to "MEDIA_FILE"
9. Mark another photo as favorite
10. Check that the tag has been added to the image file (if you used "MEDIA_FILE")

Part of #310 .
